mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -656,7 +656,8 @@ object ALS extends DefaultParamsReadable[ALS] with Logging {
         itemFactors.setName(s"itemFactors-$iter").persist(intermediateRDDStorageLevel)
         // TODO: Generalize PeriodicGraphCheckpointer and use it here.
         if (shouldCheckpoint(iter)) {
-          itemFactors.checkpoint() // itemFactors gets materialized in computeFactors.
+          // itemFactors gets materialized in computeFactors & here.
+          ALS.checkpointAndCleanParents(itemFactors)
```

Comment:
@mengxr just want to get your opinion on this - I think an extra `count` in the implicit case is not too bad.

Having said that, I took another look and it would only be a few extra lines to capture the shuffle deps in L660 & L676 (similar to how `previousCheckpointFile` works), and clean them just before `deletePreviousCheckpointFile()` is called.